File: perl/euler3.pl
#!/usr/bin/perl

$number = 600851475143;
$divisor = 2;
while( $number > 1 ) {
	if( $number % $divisor == 0 ) {
		$number /= $divisor;
		$divisor--;
	}
	$divisor++;
}

print $divisor, "\n";

File: ruby/euler3.rb
#!/usr/bin/ruby
# What is the largest prime factor of the number 600851475143 ?

number = 600851475143
divisor = 2
while number > 1 do
	if number % divisor == 0 then
		number /= divisor
		divisor -= 1
	end
	divisor += 1
end

puts(divisor)
